homeassistant/components/modbus/climate.py

```diff
@@ -252,6 +255,16 @@ def __init__(
         if CONF_HVAC_ONOFF_REGISTER in config:
             self._hvac_onoff_register = config[CONF_HVAC_ONOFF_REGISTER]
             self._hvac_onoff_write_registers = config[CONF_WRITE_REGISTERS]
+
+            if CONF_HVAC_ONOFF_REGISTER_WRITE_TYPE in config:
```

Comment:
Indeed, any default would conflict with all existing integrations
This value is optional, and only needed for those 
- controlling on/of with a coil instead of (a) register(s)
- controlling  hvac_mode in registers and hvacmode in a register, or vice versa (this comes as a side effect though)